server/test/suite_data_store.cpp

```diff
@@ -3,31 +3,331 @@
 #include <gtest/gtest.h>
 
 #include "data_store.hpp"
+#include "stdio.h"
+
+using namespace ::testing;
 
 namespace ins_service
 {
 
-class DataStoreFixture : public ::testing::Test
+class DataStoreFixture : public Test
 {
 public:
     virtual void SetUp()
     {
+        //spdlog::set_level(spdlog::level::debug);
     }
 
-    sqlite3* GetDatabase()
+    sqlite3* GetDatabase(std::shared_ptr<DataStore> ds)
     {
-        return data_store_.database_;
+        return ds->database_;
+    }
+
+    std::string GetExecutingSql()
+    {
+        return data_store_->executing_sql_;
+    }
+
+    bool CreateLocationTable()
+    {
+        return data_store_->CreateLocationTable();
+    }
+
+    bool RunQuery(std::string sql)
+    {
+        return data_store_->RunQuery(sql);
+    }
+
+    bool ReadDistinctMacAddrs(const std::string& device_id, std::vector<std::string>& mac_addrs)
+    {
+        return data_store_->ReadDistinctMacAddrs(device_id, mac_addrs);
     }
 
 protected:
-    DataStore data_store_;
+    std::shared_ptr<DataStore> data_store_ = std::make_shared<DataStore>();
 };
 
-TEST_F(DataStoreFixture, Init)
+/**
+ * TEST: Init
+ * EXPECT: Creates new db file.
+ */
+TEST_F(DataStoreFixture, Init_DbNotExisting_WillCreateNewDb)
 {
-    data_store_.Init("database_file");
-    sqlite3* db = GetDatabase();
+    data_store_->Init("database_file");
+    sqlite3* db = GetDatabase(data_store_);
     EXPECT_NE(db, nullptr);
+    std::remove("database_file");
+}
+
+/*
+ * TEST: Init
+ * EXPECT: Loads existing db
+ */
+TEST_F(DataStoreFixture, Init_DbCreated_WillLoadExistingDb)
+{
+    sqlite3* db = GetDatabase(data_store_);
+    EXPECT_EQ(db, nullptr);
+
+    data_store_->Init("database_file");
+    db = GetDatabase(data_store_);
+    EXPECT_NE(db, nullptr);
+
+    data_store_->Close();
+    db = GetDatabase(data_store_);
+    EXPECT_EQ(db, nullptr);
```

Comment:
Is this check necessary to test "the loading of existing database" or should it be a standalone/separate test?

Reply:
Hmm, was mainly thinking if this could be its own test. It is not really important just an effort to keep the tests as simple and atomic as possible.